File: Data Structures & Algorithms/max-water-container/submission-0.java
class Solution {
    public int maxArea(int[] heights) {
        int left=0, right=heights.length-1, max=0;
        while(left < right)
        {
            int sum=(right - left)* (Math.min(heights[left], heights[right]));
            if(sum > max)
            {
                  max=sum; 
            } 
                 if(heights[left] < heights[right])
                    left++;
                 else
                    right--;   
        }
        return max;
    }
}

File: Data Structures & Algorithms/split-array-largest-sum/submission-0.py
class Solution:
    def splitArray(self, nums: List[int], k: int) -> int:
        left, right = max(nums), sum(nums)
        res = right
        while left <= right:
            mid = (left + right) // 2
            count = 1
            sums = 0
            for i in nums:
                sums += i
                if sums > mid:
                    count += 1
                    sums = i  
            if count <= k:
                right = mid - 1
                res = mid
            else:
                left = mid + 1
                
        return res                        